src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -149,9 +99,199 @@ public static IResourceBuilder<SqlServerServerResource> AsAzureSqlDatabase(this
     /// <param name="builder">The builder for the SQL Server resource.</param>
     /// <param name="configureResource">Callback to configure the underlying <see cref="global::Azure.Provisioning.Sql.SqlServer"/> resource.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    [Obsolete($"This method is obsolete and will be removed in a future version. Use {nameof(AddAzureSqlServer)} instead to add an Azure SQL server resource.")]
     [Experimental("AZPROVISION001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
     public static IResourceBuilder<SqlServerServerResource> AsAzureSqlDatabase(this IResourceBuilder<SqlServerServerResource> builder, Action<IResourceBuilder<AzureSqlServerResource>, ResourceModuleConstruct, SqlServer, IEnumerable<SqlDatabase>>? configureResource)
     {
         return builder.PublishAsAzureSqlDatabase(configureResource, useProvisioner: true);
     }
+
+    /// <summary>
+    /// Adds an Azure SQL Database (server) resource to the application model.
```

Comment:
This is where the nominclature gets worse. `Azure SQL Database` is the name of the service (AFAICT). So this is the Azure SQL Database "server" resource.

I was following existing XML docs above here.

Reply:
I think colloquially people call Azure SQL Database ... Azure SQL.

Reply:
... to finish that thought. I think then it is fine to do AddAzureSqlServer(...) because effectively that is what you are doing ... creating a server resource. On the backend I seem to remember that Azure SQL Database Servers were somehow a virtual concept but things could have changed.